update: distributed authority parenting rules (#1423)
Unity-Technologies/com.unity.multiplayer.docs: docs/advanced-topics/networkobject-parenting.md

  - When `OnNetworkObjectParentChanged` is invoked, on the server side, adjust the child's transform values within the overridden method.
  - Netcode for GameObjects will then synchronize all clients with the child's parenting and transform changes.

### Parenting, Distributed Authority, and NetworkObject Redistribution

Parenting within a distributed authority network topology follows the same rules as outlined above with the minor exception that you can exchange any references to "server" with "authority". 
#### Rules Adjustments:
  - Whoever has authority of a `NewtworkObject` can control its parenting. 
  - You can parent under mixed authority. This means Client-B can parent a `NetworkObject` it has authority over under another `NetworkObject` that Client-A has authority over.

#### Distributable Permissions and Child Hierarchies:

> - **Upon a client disconnecting:**
>   - Any root parent with distributable permission set and was owned by the client that disconnected will be redistributed.
>     - If the root parent was locked when the client disconnected, then it will be unlocked prior to redistributing.
>     - All children under the root parent will "follow" the root parent ownership change only if:
>       - The child is owned by the client that disconnected.
>       - The child has either the distributable or transferable permission set.
>     - If a child's ownership permission is locked, it will be unlocked prior to redistributing.
>     - Any child of the root parent that is owned by a different client will not get redistributed.

Reviewer: ```suggestion
- **When a client disconnects:**
  - Any root parent with [distributable permission](../basics/ownership.md#ownership-permission-settings) set that was owned by the disconnected client is redistributed.
    - If the root parent was locked when the client disconnected, then it's unlocked prior to redistributing.
    - Children of the root parent change ownership with the root parent only if:
      - The child was owned by the client that disconnected.
      - The child has either the distributable or transferable permission set.
    - If a child's ownership permission is locked, it's unlocked prior to redistributing.
    - Any child of the root parent that's owned by a different client won't get redistributed.
```

> - **Upon a client connecting:**
>   - Any root parent with the distributable permission set and does not have ownership locked will be taken into consideration for redistribution.
>     - If a root parent is redistributed to a newly connecting client then:
>       - Any child under the root parent that has the same owner as the root parent will "follow" (*i.e. change ownership with the root parent*).
>         - Unless the child has ownership locked (i.e. it will not have its ownership changed).
>     - Any child under the root parent that has the distributable or transferrable permission set can be redistributed as well.
>       - Unless the child has ownership locked (i.e. it will not have its ownership changed).
> 

Reviewer: ```suggestion
- **When a client connects:**
  - Any root parent with [distributable permission](../basics/ownership.md#ownership-permission-settings) set that doesn't have ownership locked is considered for redistribution.
    - If a root parent is redistributed to a newly connecting client then:
      - Any child under the root parent that has the same owner as the root parent changes ownership with the root parent.
        - Unless the child has ownership locked.
    - Any child under the root parent that has the distributable or transferrable permission set can be redistributed as well.
      - Unless the child has ownership locked.
```
